src/Dfc.CourseDirectory.Core/ReferenceData/Ukrlp/UkrlpSyncHelper.cs

```diff
@@ -91,20 +101,25 @@ private static ProviderContact MapContact(ProviderContactStructure contact) => n
         {
             ContactAddress = new ProviderContactAddress()
             {
-                SAON = new ProviderContactAddressSAON() { Description = contact.ContactAddress.SAON.Description },
-                PAON = new ProviderContactAddressPAON() { Description = contact.ContactAddress.PAON.Description },
-                StreetDescription = contact.ContactAddress.StreetDescription,
-                Locality = contact.ContactAddress.Locality,
-                Items = contact.ContactAddress.Items,
-                PostCode = contact.ContactAddress.PostCode
+                SAON = new ProviderContactAddressSAON { Description = contact.ContactAddress.Address1 },
+                PAON = new ProviderContactAddressPAON { Description = contact.ContactAddress.Address2 },
+                StreetDescription = contact.ContactAddress.Address3,
+                Locality = contact.ContactAddress.Address4,
+                Items = new List<string>
+                {
+                    contact.ContactAddress.Town,
+                    contact.ContactAddress.County,
+                }.Where(s => s != null).ToList(),
+                PostTown = null, // town still in Items to avoid changing data after upgrade from v3 to v6 UKRLP client
+                PostCode = contact.ContactAddress.PostCode,
             },
             ContactEmail = contact.ContactEmail,
             ContactFax = contact.ContactFax,
             ContactPersonalDetails = new ProviderContactPersonalDetails()
             {
                 PersonNameTitle = contact.ContactPersonalDetails.PersonNameTitle,
                 PersonGivenName = contact.ContactPersonalDetails.PersonGivenName,
-                PersonFamilyName = contact.ContactPersonalDetails.PersonFamilyName
+                PersonFamilyName = contact.ContactPersonalDetails.PersonFamilyName,
```

Comment:
-`,`

Reply:
I always add the last comma and would suggest everyone does where permitted by the language. It avoids having git highlight unrelated lines because of having to shuffle the final comma around. It also improves automatic merge accuracy as git takes into account nearby lines when merging.

Example of adding a line when the previous developer didn't leave a trailing comma:

```patch
@@ -119,7 +119,8 @@ namespace Dfc.CourseDirectory.Core.ReferenceData.Ukrlp                                                                                                                    
             {
                 PersonNameTitle = contact.ContactPersonalDetails.PersonNameTitle,
                 PersonGivenName = contact.ContactPersonalDetails.PersonGivenName,
-                PersonFamilyName = contact.ContactPersonalDetails.PersonFamilyName                                                                                                           
+                PersonFamilyName = contact.ContactPersonalDetails.PersonFamilyName,                                                                                                          
+                PersonProperName = contact.ContactPersonalDetails.PersonProperName                                                                                                           
             },
             ContactTelephone1 = contact.ContactTelephone1,
             ContactType = contact.ContactType,
```

Example of adding a line when the previous developer left a trailing comma:

```patch
@@ -122,2 +122,3 @@ namespace Dfc.CourseDirectory.Core.ReferenceData.Ukrlp                                                                                                                    
                 PersonFamilyName = contact.ContactPersonalDetails.PersonFamilyName,
+                PersonProperName = contact.ContactPersonalDetails.PersonProperName,                                                                                                          
             },
```
Much tidier.

As per https://stackoverflow.com/questions/8253413/why-c-sharp-4-0-tolerates-trailing-comma-in-anonymous-objects-initialization-cod/8253440#8253440 and https://stackoverflow.com/questions/2147333/net-enumeration-allows-comma-in-the-last-field/32701971#32701971